pkg/patch/transformer/patchjson6902json.go

```diff
@@ -30,17 +31,31 @@ import (
 type patchJson6902JSONTransformer struct {
 	target resid.ResId
 	patch  jsonpatch.Patch
+	rawOp  []byte
 }
 
 var _ transformers.Transformer = &patchJson6902JSONTransformer{}
 
 // newPatchJson6902JSONTransformer constructs a PatchJson6902 transformer.
 func newPatchJson6902JSONTransformer(
-	id resid.ResId, p jsonpatch.Patch) (transformers.Transformer, error) {
-	if len(p) == 0 {
+	id resid.ResId, rawOp []byte) (transformers.Transformer, error) {
+	op := rawOp
+	var err error
+	if !isJsonFormat(op) {
+		// if it isn't JSON, try to parse it as YAML
+		op, err = yaml.YAMLToJSON(rawOp)
+		if err != nil {
+			return nil, err
+		}
+	}
+	decodedPatch, err := jsonpatch.DecodePatch(op)
+	if err != nil {
+		return nil, err
+	}
+	if len(decodedPatch) == 0 {
 		return transformers.NewNoOpTransformer(), nil
 	}
-	return &patchJson6902JSONTransformer{target: id, patch: p}, nil
+	return &patchJson6902JSONTransformer{target: id, patch: decodedPatch, rawOp: rawOp}, nil
 }
```

Comment:
Why move this code?

Was there going to be an improvement to the unit test for this func?

Reply:
I don't mind moving it, but since the test didn't change...

Reply:
The code of decoding patch was moved here since I want to keep `rawOp`(the content of json patch). When there is an error when patching the objects, the `rawOp` can be shown.

I'll add some tests.

Reply:
added one test